db/schema_applier.cc

```diff
@@ -912,6 +986,15 @@ void schema_applier::commit_tables_and_views() {
 }
 
 void schema_applier::commit_on_shard(replica::database& db) {
+    // commit token_metadata, must be before updating keyspaces as
+    // both may update some metadata (namely erm) but only keyspace operations
+    // include up-to-date information (because they combine keyspace change and
+    // tablet change - via pending_token_metadata).
```

Comment:
This sounds odd. Can you elaborate what metadata is updated later by keyspace operations?

Reply:
commit_token_metadata_change calls ks->update_static_effective_replication_map which overwrites _effective_replication_map

keyspace::apply is also changing _effective_replication_map